core/utils/solidity_abi.go

```diff
@@ -0,0 +1,50 @@
+package utils
+
+import (
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/gobuffalo/packr"
+	"github.com/pkg/errors"
+	"github.com/tidwall/gjson"
+)
+
+// Contract holds the solidity contract's parsed ABI
+type Contract struct {
+	ABI abi.ABI
+}
+
+func getContract(name string, box packr.Box) (*Contract, error) {
+	jsonFile, err := box.Find(name + ".json")
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to read contract JSON")
+	}
+
+	abiBytes := gjson.GetBytes(jsonFile, "compilerOutput.abi")
+	abiParsed, err := abi.JSON(strings.NewReader(abiBytes.Raw))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Contract{abiParsed}, nil
+}
+
+// GetContract loads the contract JSON file from ../evm/dist/artifacts
+// and parses the ABI JSON contents into an abi.ABI object
+func GetContract(name string) (*Contract, error) {
+	box := packr.NewBox("../../evm/dist/artifacts")
+	return getContract(name, box)
+}
+
+// GetAdvancedContract loads the contract JSON file from ../evm/v0.5/dist/artifacts
+// and parses the ABI JSON contents into an abi.ABI object
+func GetAdvancedContract(name string) (*Contract, error) {
```

Comment:
Suggestion: If the only difference between `GetContract` and `GetAdvancedContract` is the folder v0.5, maybe place that in the name? `GetContractv5`?

Reply:
Sounds good.